### DISCUSSION TITLE
pydantic base model in JSON output adaptor

### DISCUSSION THREAD

**Discussion by chlee1016** [bentoml/BentoML#2406] [Q&A] [answered]
Hi BentoML team! When I take a look at new feature in BentoML 1.0, found the pydantic model can be used in JSON I/O adaptor.
- https://docs.bentoml.org/en/latest/api/api_io_descriptors.html#bentoml.io.JSON

As I know, the Input can be checked whether it is valid using Custom pydantic model.
However, in case of the Output, even if I set the pydantic model, no action is taken.

I wonder how can the pydantic model work in JSON Output adaptor. Please let me know if I'm missing anything.
- https://github.com/bentoml/BentoML/blob/c6a50e6bfc5f4f2be7ce2e4bd793fcd776a67e9e/bentoml/_internal/io_descriptors/json.py#L171-L180

- related issue : https://github.com/bentoml/BentoML/issues/1480

**Comment by withsmilo** [comment, marked as answer]
Hi, @chlee1016 😄 
I definitely think this is a bug. The JSON IO Descriptor should validate the output json data using a pydantic model for the outputs. Please contribute to solving this before releasing v1.0 if you have time.

**Reply by aarnphm** [reply]
Thanks for spotting this out. I will take a look at this
